### PR TITLE
Add optional custom values to Gradle Build Scans

### PR DESCRIPTION
**Pull request by bigdaz**
This PR introduces 3 main changes:
1. Extracts all utility methods into a separate `Utils` class for both the Gradle Plugin and Maven Extension
2. Adds the option for `jps` and `ps` output to be added as custom values. These are enabled via system props.
3. Allows the `Test` task configuration to _not_ be added as custom values. This is disabled via system prop.

### CODE REVIEW

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import org.gradle.api.Task;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.tasks.testing.Test;
+
+import static com.gradle.Utils.*;
+
+final class CustomBuildScanConfigExtras {
+
+    static void configureBuildScan(BuildScanExtension buildScan, Gradle gradle) {
+        if (isCaptureEnabled("TestTaskConfiguration", true)) {
```
Comment (etiennestuder) [inline common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java:14]: I would set it to false by default.

Comment (etiennestuder) [reply common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java:14]: (different of today's behavior, I know)

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+class Utils {
```
Comment (etiennestuder) [inline common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java:18]: can be final

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java (hunk 1 above)
Comment (bigdaz) [reply common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java:14]: Sure. My intent was to preserve the existing functionality of the script.
Are you OK with Test task parallelism not being reported by default, too? (These are currently controlled by the same system property).

File: common-custom-user-data-maven-extension/src/main/java/com/gradle/Utils.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+class Utils {
```
Comment (etiennestuder) [inline common-custom-user-data-maven-extension/src/main/java/com/gradle/Utils.java:16]: can be final

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java (hunk 1 above)
Comment (etiennestuder) [reply common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java:14]: I think it is ok. we can still tell people in the trial navigator to set these properties to true initially, and to turn them off if it is too much noise, but by then we already have a bseline.

**Comment by etiennestuder** [comment]
Nice refactoring. My feedback is only of minor nature.

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java (hunk 2 above)
Comment (bigdaz) [reply common-custom-user-data-gradle-plugin/src/main/java/com/gradle/Utils.java]: Done

File: common-custom-user-data-maven-extension/src/main/java/com/gradle/Utils.java (hunk 3 above)
Comment (bigdaz) [reply common-custom-user-data-maven-extension/src/main/java/com/gradle/Utils.java]: Done

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java (hunk 1 above)
Comment (bigdaz) [reply common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanConfigExtras.java]: Done

**Pull request merged**